### ISSUE TITLE
Extending a template from a different crate

### ISSUE THREAD

**Issue by tecywiz121** [djc/askama#184]
Is it possible with askama to extend a template from another crate?

I'm trying to split parts of my web application into different "plugins", but I'd like there to be a common ancestor template for a consistent look and feel.

**Comment by djc** [comment]
Askama currently needs a path relative to the current template or to the crate's template dir. So if you have all your crates in a single repo/workspace, you can probably make it work.

**Comment by tecywiz121** [comment]
Would that still work when uploaded to crates.io?

**Comment by djc** [comment]
Hmm, probably not.

**Comment by djc** [comment]
Hmm, I think it might still work if you use the old inheritance code path (with an explicit parent field), but that was supposed to be deprecated soon.

**Comment by tecywiz121** [comment]
Is that still documented somewhere?

**Comment by djc** [comment]
Hmm, I guess not.

**Comment by tecywiz121** [comment]
Since `_parent` is on the deprecation path, would adding a new sub-attribute `extends = other_crate::templates::FooBar` make sense?

Something like:

```rust
use other_crate;

#[derive(Template)]
#[template(path = "types/detail.html", extends = other_crate::templates::Base)]
pub struct Detail {
    some_value: String
}
```

```jinja2
{% extends %}{# Keep some kind of "extends" tag to show that this isn't a complete template #}

{% block base_body %}
	{{ some_value }}
{% endblock %}
```

**Comment by djc** [comment]
I've decided that I do want to support this use case, and I think I've come up with a decent design for it. Are you interested in helping to implement that?

**Comment by tecywiz121** [comment]
I think I'd be willing to try my hand at it, but I'd like to hear more about your design before I commit to anything, if that's alright.

**Comment by tecywiz121** [comment]
Okay, that sounds like it would work for my use case. If I get the chance to work on it, I'll let you know before I start.

**Comment by tecywiz121** [comment]
I've started taking a look at this finally. I have a couple questions, if you wouldn't mind mentoring me a bit.

Implementing `Deref` on the child template for the named field isn't too bad, but I wanted to double check that's still a good thing to do.

Is the trait you mentioned a new one? I assume so. I _think_ what I need to do is to add a `Context` for the named parent, correct?

**Comment by djc** [comment]
Yeah, so I'm afraid this is harder than I made it look previously: I removed a whole bunch of that code generation earlier and only left the part that deals with dereferencing through the `_parent` field.

I don't think you need to add a `Context` for the named parent, but you'd need to generate a trait implementation with a method that takes a trait object implementing the trait, and then you need to make sure this works with deeper in heritance.

As an alternative, have you considered leveraging the idea that `Display` is implemented for each implementer of `Template` and that `Display` fields can trivially be used directly in templates to defer to another type's template implementation? That might provide a much easier way forward for your use case (although it doesn't enable the full flexibility).
